internal/repositories/urls/sql_repository.go

```diff
@@ -0,0 +1,92 @@
+package urls
+
+import (
+	"context"
+	"time"
+
+	"github.com/izaake/go-shortener-tpl/internal/models"
+	"github.com/izaake/go-shortener-tpl/internal/storage"
+)
+
+type sqlRepository struct {
+	s storage.Storage
+}
+
+func NewSQLRepository(s storage.Storage) Repository {
+	return &sqlRepository{s: s}
+}
+
+func (r sqlRepository) Save(user *models.User, ignoreConflicts bool) error {
```

Comment:
Вообще в эти функции необходимо пробрасывать вполне определенный контекст - и это контекст запроса, получаемый в хендлере через r.Context(). В этом случае при прерывании http-запроса будет прерван и sql-запрос. 
```suggestion
func (r sqlRepository) Save(ctx context.Context, user *models.User, ignoreConflicts bool) error {
```

Потребуется модификация интерфейса и всех функций всех репозиториев, но для файлового и in-memory можно использовать заглушку 
```
func (r memoryRepository) Save(_ context.Context, user *models.User, ignoreConflicts bool) error {
```